OutOfSchool/OutOfSchool.WebApi/Hubs/ChatWorkshopHub.cs

```diff
@@ -0,0 +1,252 @@
+using Newtonsoft.Json;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Models.ChatWorkshop;
+using OutOfSchool.WebApi.Services;
+using Serilog;
+
+namespace OutOfSchool.WebApi.Hubs
+{
+    [Authorize(AuthenticationSchemes = "Bearer")]
+    [Authorize(Roles = "provider,parent")]
+    public class ChatWorkshopHub : Hub
+    {
+        // This collection tracks users with their connections.
+        private static readonly ConcurrentDictionary<string, HashSet<string>> UsersConnections
+            = new ConcurrentDictionary<string, HashSet<string>>(StringComparer.InvariantCultureIgnoreCase);
+
+        private readonly ILogger logger;
+        private readonly IChatMessageWorkshopService messageService;
+        private readonly IChatRoomWorkshopService roomService;
+        private readonly IValidationService validationService;
+        private readonly IWorkshopRepository workshopRepository;
+        private readonly IParentRepository parentRepository;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChatWorkshopHub"/> class.
+        /// </summary>
+        /// <param name="chatMessageService">Service for ChatMessage entities.</param>
+        /// <param name="chatRoomService">Service for ChatRoom entities.</param>
+        /// <param name="validationService">Service for validation parameters.</param>
+        /// <param name="workshopRepository">Repository for workshop entities.</param>
+        /// <param name="parentRepository">Repository for parent entities.</param>
+        /// <param name="logger">Logger.</param>
+        public ChatWorkshopHub(ILogger logger, IChatMessageWorkshopService chatMessageService, IChatRoomWorkshopService chatRoomService, IValidationService validationService, IWorkshopRepository workshopRepository, IParentRepository parentRepository)
+        {
+            this.logger = logger;
+            this.messageService = chatMessageService;
+            this.roomService = chatRoomService;
+            this.validationService = validationService;
+            this.workshopRepository = workshopRepository;
+            this.parentRepository = parentRepository;
+        }
+
+        public override async Task OnConnectedAsync()
+        {
+            var userId = Context.User.FindFirst("sub")?.Value;
+            var userRole = Context.User.FindFirst("role")?.Value;
+
+            if (userId is null || userRole is null)
+            {
+                throw new ArgumentException($"{nameof(userId)} or {nameof(userRole)}");
+            }
+
+            logger.Information($"New Hub-connection established. UserId: {userId}");
+
+            this.AddUsersConnectionIdTracking(userId);
+
+            // Add User to all Groups where he is a member.
+            IEnumerable<long> usersRoomIds;
+            bool userRoleIsProvider = userRole.Equals(Role.Provider.ToString(), StringComparison.OrdinalIgnoreCase);
+            if (userRoleIsProvider)
```

Comment:
`if` can be simplified when call to the `GetEntityIdAccordingToUserRoleAsync` will be out of it

Reply:
yes, but isn't it easier to read the code when you see the name of the condition?

Reply:
I believe that less code here will be more readable.
Also if you call the same method with the same variables twice - both results should contain the same value, why variable should have a different name ?